src/cargo/core/gc.rs

```diff
@@ -0,0 +1,550 @@
+        }
+        Ok(())
+    }
+
+    /// Performs garbage collection based on the given options.
+    pub fn gc(
+        &mut self,
+        clean_ctx: &mut CleanContext<'config>,
+        gc_opts: &GcOpts,
+    ) -> CargoResult<()> {
+        self.global_cache_tracker.clean(clean_ctx, gc_opts)?;
+        // In the future, other gc operations go here, such as target cleaning.
+        Ok(())
+    }
+}
+
+/// Returns the shorter duration from `cur_span` versus `config_span`.
+///
+/// This is used because the user may specify multiple options which overlap,
+/// and this will pick whichever one is shorter.
+///
+/// * `cur_span` is the span we are comparing against (the value from the CLI
+///   option). If None, just returns the config duration.
+/// * `config_name` is the name of the config option the span is loaded from.
+/// * `config_span` is the span value loaded from config.
+fn newer_time_span_for_config(
+    cur_span: Option<Duration>,
+    config_name: &str,
+    config_span: &str,
+) -> CargoResult<Option<Duration>> {
+    let config_span = parse_time_span_for_config(config_name, config_span)?;
+    Ok(Some(maybe_newer_span(config_span, cur_span)))
+}
+
+/// Returns whichever [`Duration`] is shorter.
+fn maybe_newer_span(a: Duration, b: Option<Duration>) -> Duration {
+    match b {
+        Some(b) => {
+            if b < a {
+                b
+            } else {
+                a
+            }
+        }
+        None => a,
+    }
+}
+
+/// Parses a frequency string.
+///
+/// Returns `Ok(None)` if the frequency is "never".
+fn parse_frequency(frequency: &str) -> CargoResult<Option<Duration>> {
+    if frequency == "always" {
+        return Ok(Some(Duration::new(0, 0)));
+    } else if frequency == "never" {
+        return Ok(None);
+    }
+    let duration = maybe_parse_time_span(frequency).ok_or_else(|| {
+        format_err!(
+            "config option `gc.auto.frequency` expected a value of \"always\", \"never\", \
+             or \"N seconds/minutes/days/weeks/months\", got: {frequency:?}"
+        )
+    })?;
+    Ok(Some(duration))
+}
+
+/// Parses a time span value fetched from config.
+///
+/// This is here to provide better error messages specific to reading from
+/// config.
+fn parse_time_span_for_config(config_name: &str, span: &str) -> CargoResult<Duration> {
+    maybe_parse_time_span(span).ok_or_else(|| {
+        format_err!(
+            "config option `{config_name}` expected a value of the form \
+             \"N seconds/minutes/days/weeks/months\", got: {span:?}"
+        )
+    })
+}
+
+/// Parses a time span string.
+///
+/// Returns None if the value is not valid. See [`parse_time_span`] if you
+/// need a variant that generates an error message.
+fn maybe_parse_time_span(span: &str) -> Option<Duration> {
+    let Some(right_i) = span.find(|c: char| !c.is_ascii_digit()) else {
+        return None;
+    };
+    let left = &span[..right_i];
+    let mut right = &span[right_i..];
```

Comment:
We could use [str::split_at](https://doc.rust-lang.org/std/string/struct.String.html#method.split_at)